src/compiler/checker.ts

```diff
@@ -12901,13 +12901,18 @@ namespace ts {
             return result & ObjectFlags.PropagatingFlags;
         }
 
+        function containsMarkerType(typeArguments: readonly Type[] | undefined) {
+            return some(typeArguments, t => t === markerSuperType || t === markerSubType || t === markerOtherType);

```

Comment:
This check happens a few times - maybe just hoist it out to its own function and call it in `reportUnmeasurableMarkers`/`reportUnreliableMarkers`.